src/internal/RangeCounting.cpp

```diff
@@ -0,0 +1,286 @@
+#include <tvm/internal/RangeCounting.h>
+
+namespace tvm::internal
+{
+bool RangeCounting::add(const Range & r)
+{
+  assert(isValid());
+  if(r.dim == 0)
+    return false;
+
+  recomputeSplit_ = true;
+  if(limits_.empty())
+  {
+    limits_.emplace_back(r.start, Limit::Lower);
+    limits_.emplace_back(r.end(), Limit::Upper);
+    return recompute(true);
+  }
+  else
+  {
+    int depth = 0;
+    bool change = false;
+    Limit l = {r.start, Limit::Lower};
+    Limit u = {r.end(), Limit::Upper};
+    auto it = limits_.begin();
+    moveToFirstAfter(l, it, depth);
+    if(it == limits_.end())
+    {
+      limits_.push_back(l);
+      limits_.push_back(u);
+      return recompute(true);
+    }
+    insert(l, it, depth);
+    if(depth == 1)
+      change = true;
+
+    change = moveToFirstAfter(Limit(r.end(), Limit::Cut), it, depth, 1) || change;
```

Comment:
I don't understand why `1` here?

Reply:
We are adding a range `[l,u)`.
We are interested in detecting if this range covers portions with depth 0. Since we already added `l`, those portions now have depth one.

I added a comment in 32599c3